elyra/pipeline/handlers.py

```diff
@@ -174,6 +174,29 @@ async def get(self, processor, component_id):
         self.finish(properties_json)
 
 
+class PipelineComponentDefinitionHandler(HttpErrorMixin, APIHandler):
+    """Handler to expose method calls to retrieve the definition for a given component_id"""
+
+    @web.authenticated
+    async def get(self, processor, component_id):
+        self.log.debug(f'Retrieving component definition for component: {component_id}')
+
+        if PipelineProcessorManager.instance().is_supported_runtime(processor) is False:
+            raise web.HTTPError(400, f"Invalid processor name '{processor}'")
+
+        if not component_id:
+            raise web.HTTPError(400, "Missing component ID")
+
+        component: Optional[Component] = \
+            await PipelineProcessorManager.instance().get_component(processor, component_id)
```

Comment:
I think this will return `None` if the `component_id` can't be found, so we should check for that and raise a 404. (We should probably consider raising a NotFound exception from the cache instead and include the id in the message, etc. - for another day.)

The 400 raised if the component_id is missing entirely seems like a good match.